core/src/main/scala/org/apache/spark/sql/catalyst/planner/TiAggregationImpl.scala

```diff
@@ -84,15 +84,11 @@ object TiAggregationImpl {
               a.copy(aggregateFunction = Count(r.child), resultId = newExprId))
         }.toMap
 
+        // ResultDataType of sum only can be double or decimal
+        // We need to cast decimal back to long for the promotedSum, or spark will regard decimal as long cause the wrong answer
         val sumRewrite = sumsRewriteMap.map {
           case (ref, sum) =>
-            val resultDataType = sum.resultAttribute.dataType
-            val castedSum = resultDataType match {
-              case LongType => Cast(sum.resultAttribute, DecimalType.BigIntDecimal)
-              case FloatType => Cast(sum.resultAttribute, DoubleType)
```

Comment:
Should we keep the float case

Reply:
the float type will not be pushed down. but I think your suggestion is better in case float  can pushdown in some day